src/interceptors/WebSocket/WebSocketTransport.ts

```diff
@@ -0,0 +1,36 @@
+export type WebSocketSendData =
+  | string
+  | ArrayBufferLike
+  | Blob
+  | ArrayBufferView
+
+export type WebSocketTransportOnIncomingCallback = (
+  event: MessageEvent<WebSocketSendData>
+) => void
+
+export type WebSocketTransportOnOutgoingCallback = (
+  data: WebSocketSendData
+) => void
+
+export abstract class WebSocketTransport {
```

Comment:
Transport is an abstract class responsible for handling incoming and outgoing events from a WebSocket instance. 

At its core, the transport is the barebones implementation of how to apply `client.send()` events and how to listen to `client.on()` events.

Later, the transport is wrapped in the Connection instance to acquire a nicer public API.